### PR TITLE
Fix all broken links/permanent redirects/removed anchors

### PR DESCRIPTION
**Pull request by wouterj**
Today, I discovered that Sphinx has a built-in link checker. I couldn't resist running it and fixing everything it found.

It turns out that we had a lot of broken/permanently redirected links.

| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.3+ |
| Fixed tickets | - |

### CODE REVIEW

File: book/controller.rst [hunk 1]
```diff
@@ -429,9 +429,7 @@ A great way to see the core functionality in action is to look in the
 Redirecting
 ~~~~~~~~~~~
 
-If you want to redirect the user to another page, use the
-:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::redirect`
```
Comment (WouterJ) [inline book/controller.rst]: Controller API method links are removed as these methods are protected. Only public methods are included in the online API doc.

File: components/http_foundation/sessions.rst [hunk 2]
```diff
@@ -226,9 +226,6 @@ has a simple API
 :method:`Symfony\\Component\\HttpFoundation\\Session\\Attribute\\AttributeBagInterface::has`
     Returns true if the attribute exists.
 
-:method:`Symfony\\Component\\HttpFoundation\\Session\\Attribute\\AttributeBagInterface::keys`
-    Returns an array of stored attribute keys.
```
Comment (WouterJ) [inline components/http_foundation/sessions.rst]: Seems like this method never existed (or was removed before a release).

File: book/controller.rst (hunk 1 above)
Comment (stof) [inline book/controller.rst]: this should be changed IMO. Protected methods are extension point, so they should be documented in the API doc too (private methods should indeed not appear)

Comment (xabbuh) [inline book/controller.rst]: I agree with @stof and IIRC we have/had an issue for @fabpot about this.

**Comment by stof** [comment]
Is it possible to make this link checker run on Travis ?

File: book/controller.rst [hunk 3]
```diff
@@ -750,10 +748,9 @@ Forwarding to Another Controller
 --------------------------------
 
 Though not very common, you can also forward to another controller internally
-with the :method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::forward`
-method. Instead of redirecting the user's browser, it makes an internal sub-request,
-and calls the controller. The ``forward()`` method returns the ``Response``
-object that's returned from *that* controller::
+with the ``forward()`` method. Instead of redirecting the user's browser, it
+makes an internal sub-request, and calls the controller. The ``forward()``
```
Comment (javiereguiluz) [inline book/controller.rst]: Should we remove the comma before the `and`?

File: book/doctrine.rst [hunk 4]
```diff
@@ -519,11 +519,10 @@ of the bundle::
...
 
     This article shows working with Doctrine from within a controller by using
-    the :method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::getDoctrine`
-    method of the controller. This method is a shortcut to get the
-    ``doctrine`` service. You can work with Doctrine anywhere else
-    by injecting that service in the service. See
-    :doc:`/book/service_container` for more on creating your own services.
+    the ``getDoctrine()`` method of the controller. This method is a shortcut
+    to get the ``doctrine`` service. You can work with Doctrine anywhere else
+    by injecting that service in the service. See :doc:`/book/service_container`
```
Comment (javiereguiluz) [inline book/doctrine.rst]: This should be reworded to improve its readability: `by injecting that service in the service`

File: cookbook/controller/service.rst [hunk 5]
```diff
@@ -240,22 +240,22 @@ Base Controller Methods and Their Service Replacements
 This list explains how to replace the convenience methods of the base
 controller:
 
-:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::createForm` (service: ``form.factory``)
+``createForm()`` (service: ``form.factory``)
```
Comment (javiereguiluz) [inline cookbook/controller/service.rst]: I'm curious: what's the reason for removing the links to the base `Controller` methods?

**Comment by javiereguiluz** [comment]
:+1: thanks a lot for fixing all these errors!

File: book/controller.rst (hunk 3 above)
Comment (WouterJ) [inline book/controller.rst]: yes, but let's do that in the docbot PRs and keep this one focussed on URLs.

File: cookbook/controller/service.rst (hunk 5 above)
Comment (WouterJ) [inline cookbook/controller/service.rst]: See https://github.com/symfony/symfony-docs/pull/5553#discussion_r35249678

**Comment by WouterJ** [comment]
@stof it is possible (it's quite easy, as it's a built-in sphinx builder). However, I don't think it's worth it, as there are many URLs in the Symfony documentation (so running this builder takes some minutes) and not a lot of PRs add new external references. I want to provide a PR status to the contributor as quick as possible, so I don't think making the travis build longer is really needed here.

I thought about only including it in push Travis builds (and not PRs updates). However, we don't get emails about failing builds (and least, I don't get them) and I don't think we'll regularly check Travis to see the status of the branch.

**Comment by stof** [comment]
@WouterJ for push builds, the one receiving the email is the dev doing the push, which means the one doing the merge of the PR.

**Comment by WouterJ** [comment]
Reverted the removal of all API method links to protected methods, I agree that it should be fixed in the API generator instead.

Ready to merge imo

(fyi, @stof, we decided against adding this to the build during the doc meeting).

File: contributing/documentation/translations.rst [hunk 6]
```diff
@@ -89,4 +89,4 @@ repository and apply changes to the translated documents as soon as possible.
     repositories as obsolete documentation is dangerous.
 
 .. _`an ongoing discussion`: https://github.com/symfony/symfony-docs/issues/4078
-.. _Symfony docs mailing-list: http://groups.google.com/group/symfony-docs
+.. _Symfony docs mailing-list: https://groups.google.com/forum/#!forum/symfony-docs
```
Comment (fabpot) [inline contributing/documentation/translations.rst]: This one should probably be removed as it has been disabled for quite some time now.

**Comment by WouterJ** [comment]
I've removed the mailing list link in 185dfbf, as noticed by @fabpot. /cc @weaverryan @xabbuh can you please review & vote?

**Comment by weaverryan** [comment]
I just scanned through - looks like all good stuff to me! :+1:

**Pull request merged**